osu.Game/Overlays/BeatmapListing/BeatmapSearchMultipleSelectionFilterRow.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input.Events;
+using osuTK;
+
+namespace osu.Game.Overlays.BeatmapListing
+{
+    public abstract class BeatmapSearchMultipleSelectionFilterRow<T> : BeatmapSearchFilterRow<List<T>>
+    {
+        protected BeatmapSearchMultipleSelectionFilterRow(string headerName)
+            : base(headerName)
+        {
+        }
+
+        protected override Drawable CreateFilter() => CreateMultipleSelectionFilter();
+
+        protected abstract MultipleSelectionFilter CreateMultipleSelectionFilter();
+
+        protected abstract class MultipleSelectionFilter : FillFlowContainer<MultipleSelectionFilterTabItem>, IHasCurrentValue<List<T>>
+        {
+            private readonly BindableWithCurrent<List<T>> current = new BindableWithCurrent<List<T>>();
+
+            public Bindable<List<T>> Current
+            {
+                get => current.Current;
+                set => current.Current = value;
+            }
```

Comment:
`Bindable<List<T>>` is not done anywhere. `BindableList<T>` exists in framework and has proper handling of callbacks on add/remove/replace/move/reset operations. You wouldn't have to do this whole rigamarole of reconstructing the entire list to trigger a value change if you used it.